Sources/main.cpp

```diff
@@ -83,19 +84,35 @@ CreateSocket(std::string const &host, std::string const &port, bool reverse) {
 
 #if !defined(OS_WIN32)
 static int PlatformMain(std::string const &host, std::string const &port) {
-  std::unique_ptr<Socket> server = CreateSocket(host, port, false);
+  struct PlatformClient {
+    std::unique_ptr<Socket> socket;
+    PlatformSessionImpl impl;
+    Session session;
+
+    PlatformClient(std::unique_ptr<Socket> socket_)
+        : socket(std::move(socket_)),
+          session(ds2::GDBRemote::kCompatibilityModeLLDB) {
+      session.setDelegate(&impl);
+      session.create(socket.get());
+    }
+  };
 
-  PlatformSessionImpl impl;
+  std::unique_ptr<Socket> serverSocket = CreateSocket(host, port, false);
 
   do {
-    std::unique_ptr<Socket> client = server->accept();
-    // Platform mode implies that we are talking to an LLDB remote.
-    Session session(ds2::GDBRemote::kCompatibilityModeLLDB);
-    session.setDelegate(&impl);
-    session.create(client.get());
-
-    while (session.receive(/*cooked=*/false))
-      continue;
+    std::unique_ptr<Socket> clientSocket = serverSocket->accept();
+    auto platformClient =
+        ds2::make_unique<PlatformClient>(std::move(clientSocket));
```

Comment:
Why not create this inside the thread? Error handling?

Why use a unique_ptr? Is one of the members of PlatformClient not movable?